test/unit/deferred.js

```diff
@@ -167,6 +167,42 @@ test( "jQuery.Deferred.then - filtering (fail)", function( assert ) {
 	});
 });
 
+test( "jQuery.Deferred.catch", function( assert ) {
+	assert.expect( 4 );
+
+	var value1, value2, value3,
+		defer = jQuery.Deferred(),
+		piped = defer.catch(function( a, b ) {
+			return a * b;
+		}),
+		done = jQuery.map( new Array( 3 ), function() { return assert.async(); } );
+
+	piped.done(function( result ) {
+		value3 = result;
+	});
+
+	defer.fail(function( a, b ) {
+		value1 = a;
+		value2 = b;
+	});
+
+	defer.reject( 2, 3 ).catch(function() {
+		assert.strictEqual( value1, 2, "first reject value ok" );
+		assert.strictEqual( value2, 3, "second reject value ok" );
+		assert.strictEqual( value3, 6, "result of filter ok" );
+		done.pop().call();
+	});
+
+	jQuery.Deferred().resolve().catch(function() {
```

Comment:
I remember some recent tweets from @rwaldron "advocating" to use airbnb - https://twitter.com/search?q=idiomatic%20airbnb%20from%3Arwaldron&src=typd&lang=en instead of idiomatic.

> There are a bunch of things in the Airbnb guide that apply to ES6+ and it will probably be a while before we need those

Well, i remember there was some discussion to use ES6 `import` instead of `amd` and who knows, maybe jQuery written on ES6 would be actually smaller after transpiling and minifying :-), we didn't yet try that :0)